src/pages/questions/ProductInformation.jsx

```diff
@@ -130,6 +150,19 @@ const ProductInformation = ({ question }) => {
                   style={{ maxWidth: 300, maxHeight: 300 }}
                 />
               </Zoom>
+              {flagged[src[src.length - 5]] ? (
+                <FlagIcon
+                  onClick={() =>
+                    deleteFlagImage(src, question.barcode, src[src.length - 5])
```

Comment:
what about passing the `src` and `question.barcode` to `deleteFlagImage` , and in `deleteFlagImage` obtain the `index` (imageId?), and that same `index` can be passed down to `updateFlagImage()` which seems to be needing it.

Reply:
the helper `getImagesUrls` only return image src, but it can be modified to return, all the necessary information. Especially imageId which will avoid doing the `src[src.length - 5]`

Reply:
Is there a better way to get the imgid than `src[src.length - 5]`? Since it is not handling the double digit cases?

Reply:
what about splitting, taking the last part and then removing the file extension, something like:
`src = "https://static.openfoodfacts.org/images/products/004/900/053/2258/1.jpg"`
`src2= src.split("/").at(-1)`
`src2.replace(/\..+$/, "")` or maybe removing the non digits?

please whenever you use it, please add a comment like
`// src looks like: "https://static.openfoodfacts.org/images/products/004/900/053/2258/1.jpg" `
that way we don't have to remember how the src was